controllers/spifilecontentrequest_controller.go

```diff
@@ -105,11 +120,40 @@ func (r *SPIFileContentRequestReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, fmt.Errorf("failed to read the object: %w", err)
 	}
 
+	finalizationResult, err := r.finalizers.Finalize(ctx, &request)
+	if err != nil {
+		// if the finalization fails, the finalizer stays in place, and so we don't want any repeated attempts until
+		// we get another reconciliation due to cluster state change
+		return ctrl.Result{Requeue: false}, fmt.Errorf("failed to finalize: %w", err)
+	}
+	if finalizationResult.Updated {
+		if err = r.K8sClient.Update(ctx, &request); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update file request based on finalization result: %w", err)
+		}
+	}
+	if finalizationResult.StatusUpdated {
+		if err = r.K8sClient.Status().Update(ctx, &request); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to update the file request status based on finalization result: %w", err)
+		}
+	}
+
 	if request.DeletionTimestamp != nil {
 		lg.Info("file request object is being deleted")
 		return ctrl.Result{}, nil
 	}
 
+	// cleanup file content requests by lifetime
+	requestLifetime := time.Since(request.CreationTimestamp.Time).Seconds()
+	if requestLifetime > r.Configuration.FileContentRequestTtl.Seconds() {
```

Comment:
I think using `time.After` or `time.Before` would be more readable https://docs.studygolang.com/pkg/time/#Time.After. We're doing similar thing with SPIAccessCheck https://github.com/redhat-appstudio/service-provider-integration-operator/blob/main/controllers/spiaccesscheck_controller.go#L68